common/fe-ste-models.go

```diff
@@ -591,15 +598,23 @@ func (ft *FromTo) From() Location {
 }
 
 func (ft *FromTo) IsDownload() bool {
-	return ft.From().IsRemote() && ft.To().IsLocal()
+	return ft.From().IsRemote() && ft.To().IsLocal() && ft.To() != ELocation.None() && ft.To() != ELocation.Unknown()
 }
 
 func (ft *FromTo) IsS2S() bool {
-	return ft.From().IsRemote() && ft.To().IsRemote()
+	return ft.From().IsRemote() && ft.To().IsRemote() && ft.To() != ELocation.None() && ft.To() != ELocation.Unknown()
 }
 
 func (ft *FromTo) IsUpload() bool {
-	return ft.From().IsLocal() && ft.To().IsRemote()
+	return ft.From().IsLocal() && ft.To().IsRemote() && ft.To() != ELocation.None() && ft.To() != ELocation.Unknown()
+}
+
+func (ft *FromTo) IsDelete() bool {
+	return ft.To() == ELocation.Unknown()
+}
+
+func (ft *FromTo) IsSetProperties() bool {
```

Comment:
Should this check that From isnt S3 or GCP? Same with Delete

Reply:
We won't permit set properties this way anyway as they're not registered in the FromTo Enum.